drake/solvers/NloptSolver.cpp

```diff
@@ -0,0 +1,269 @@
+
+#include <iostream>
+#include <stdexcept>
+#include <list>
+#include <vector>
+
+#include <nlopt.hpp>
+
+#include "NloptSolver.h"
+#include "Optimization.h"
+
+namespace Drake {
+namespace {
+Eigen::VectorXd MakeEigenVector(const std::vector<double>& x) {
+  Eigen::VectorXd xvec(x.size());
+  for (size_t i = 0; i < x.size(); i++) {
+    xvec[i] = x[i];
+  }
+  return xvec;
+}
+
+TaylorVecXd MakeInputTaylorVec(const Eigen::VectorXd& xvec,
+                               const VariableList& variable_list) {
+  size_t vars = 0;
+  for (const DecisionVariableView& v : variable_list) {
+    vars += v.size();
+  }
+
+  auto tx = initializeAutoDiff(xvec);
+  TaylorVecXd this_x(vars);
+  size_t index = 0;
+  for (const DecisionVariableView& v : variable_list) {
+      this_x.segment(index, v.size()) = tx.segment(v.index(), v.size());
+      index += v.size();
+    }
+  return this_x;
+}
+
+double EvaluateCosts(const std::vector<double>& x,
+                     std::vector<double>& grad,
+                     void* f_data) {
+  const OptimizationProblem* prog =
```

Comment:
It's the same name which is used in SnoptWrapper, LeastSquares::solve(), and testOptimizationProblem (pervasively).   Which is not to say it makes a ton of sense, I guess I could change it if we wanted too.